File: mqttsnclient/src/main/java/net/farsystem/mqttsngatek/mqttsnclient/MQTTSNConnAck.java
package net.farsystem.mqttsngatek.mqttsnclient;

public class MQTTSNConnAck {
    private final int returnCode;
    public MQTTSNConnAck(int returnCode) {
        this.returnCode = returnCode;
    }
    public int getReturnCode() {
        return returnCode;
    }
}

File: src/main/kotlin/net/farsystem/mqttsngatek/MQTTSNConnack.kt
package net.farsystem.mqttsngatek

import java.nio.ByteBuffer

data class MQTTSNConnack(val returnCode: MQTTSNReturnCode): MQTTSNBody {
    override fun writeTo(buffer: ByteBuffer): ByteBuffer = buffer.put(returnCode.code.toByte())
    override fun length(): Int = 1
}